Attachment's save button click event is now always named "save"

squelart opened the pull request:
#### 36c736481d50204339ae930b9922c6a03bb06aae
<pre>
Attachment&apos;s save button click event is now always named &quot;save&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=253269">https://bugs.webkit.org/show_bug.cgi?id=253269</a>
rdar://106108684

Reviewed by Tim Nguyen.

The HTML attribute &quot;save&quot; doesn&apos;t need a value anymore. If it&apos;s present with any value, the save button is shown, and clicking on it will dispatch a MouseEvent with type name &quot;save&quot;.

* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::saveAtom):
(WebCore::saveAttr):
(WebCore::HTMLAttachmentElement::ensureModernShadowTree):
(WebCore::HTMLAttachmentElement::updateSaveButton):
(WebCore::HTMLAttachmentElement::parseAttribute):
* Source/WebCore/html/HTMLAttachmentElement.h:

Canonical link: <a href="https://commits.webkit.org/261206@main">https://commits.webkit.org/261206@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/520ef910c51c74db66d1dc9ee9b1b43ff13958e8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110625 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19705 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43202 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1967 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119476 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114584 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21118 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10824 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/1500 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116366 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15724 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98913 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102919 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97685 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30580 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44036 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12353 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31918 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/86625 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12905 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8875 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18270 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51535 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7776 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14845 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/html/HTMLAttachmentElement.cpp:
```diff
@@ -137,9 +137,15 @@ static QualifiedName subtitleAttr()
     return QualifiedName { nullAtom(), "subtitle"_s, nullAtom() };
 }
 
+static const AtomString& saveEvent()
```
You can name this `saveAtom()`, it would be less odd especially since you're using it in `saveAttr()`

nt1m commented on Source/WebCore/html/HTMLAttachmentElement.cpp:
```diff
@@ -227,12 +232,12 @@ class AttachmentSaveEventListener final : public EventListener {
 };
 
-void HTMLAttachmentElement::updateSaveButton(const AtomString& eventTypeName)
+void HTMLAttachmentElement::showSaveButton(bool show)
```
`toggleSaveButton` ?

squelart replied:
As discuss, I'll revert to `updateSaveButton`.

squelart committed:
Attachment's save button click event is now always named "save"
https://bugs.webkit.org/show_bug.cgi?id=253269
rdar://106108684

Reviewed by Tim Nguyen.

The HTML attribute "save" doesn't need a value anymore. If it's present with any value, the save button is shown, and clicking on it will dispatch a MouseEvent with type name "save".

* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::saveAtom):
(WebCore::saveAttr):
(WebCore::HTMLAttachmentElement::ensureModernShadowTree):
(WebCore::HTMLAttachmentElement::updateSaveButton):
(WebCore::HTMLAttachmentElement::parseAttribute):
* Source/WebCore/html/HTMLAttachmentElement.h:

Canonical link: https://commits.webkit.org/261206@main
